lib/aurora_bootstrapper/notifier.rb

```diff
@@ -12,7 +12,48 @@ def export_date
     
     # ENV is string to string dictionary
     def export_date_override
-      DateTime.now.strftime("%Y-%m-%d") if ENV.key?('EXPORT_DATE_OVERRIDE')
+      date_override = nil
+
+      if ENV.key?('EXPORT_DATE_OVERRIDE')
+        now = Date.today
+
+        # expiration time is 30 days
+        for i in 1..30
+          if exists_db_dump_done_for_one_day?( date: (now-i))
+            date_override = convert_date_to_string(date: (now-i))
+            break
+          end
+        end
+
+        # the first run for the given db partition
+        if date_override.nil?
+          date_override = convert_date_to_string(date: now)
+        end
+      end
+
+      date_override
```

Comment:
Can we short circuit some of these break statements, for ease of readability like

```ruby
class Notifier
  DATE_FORMAT = "%Y-%m-%d"
  
  def export_date_override
    return unless ENV.key?('EXPORT_DATE_OVERRIDE')
    
    days_ago = (1..30).find do | days_ago |
      exists_export? date: Date.today-days_ago
    end
    
    ( Date.today - days_ago.to_i ).strftime( DATE_FORMAT )
  end
end
```

Reply:
I think I made a bug here: the datetime folder should be now-i+1 instead of now-i assuming the folder now-i contains DONE